### PR TITLE
fix(deprecation): move from deprecated listener to compatible hook system in PHPUnit 8

### PR DESCRIPTION
**Pull request by vlaurier**
Closes https://github.com/johnkary/phpunit-speedtrap/issues/60

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ matrix:
     env: PHPUNIT=8.*
   - php: 7.1
     env: PHPUNIT=dev-master MIN_STABILITY=dev
+  - php: 7.2
```
Comment (vlaurier) [inline .travis.yml]: I have excluded this failed check because PHPUnit now required PHP 7.3 at least.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,18 @@ SpeedTrap is installable via [Composer](http://getcomposer.org) and should be ad
 
 Enable with all defaults by adding the following code to your project's `phpunit.xml` file:
 
+If you are using PHPUnit 8 or above, and phpnit-speedtrap >= XXX, you should register the listener like this:
```
Comment (vlaurier) [inline README.md]: To be decided which phpunit-speedtrap it will be.

**Comment by mremi** [comment]
Hi @johnkary can you have a look please?

**Comment by johnkary** [comment]
@vlaurier I have published a new v3.2.0 that updates the Travis CI matrix. Could you merge master into this branch, resolve conflicts and push again to this PR?

This new implementation will be the basis for SpeedTrap v4.0!

**Comment by vlaurier** [comment]
@johnkary @mremi 
It should be good that way. Notice I also removed PHPUnit 7 support in composer.json as these changes are not BC compatible.

**Comment by vlaurier** [comment]
@johnkary , are you OK with these changes then ?

**Comment by thePanz** [comment]
Any updates @johnkary ?

**Comment by angelsk** [comment]
Any ETA on this merge?  Can I do anything to help test?

**Comment by VaN-dev** [comment]
I would need this to be merged too.

**Comment by johnkary** [comment]
Thank you, @vlaurier. I will have a look soon and merge toward a v4.0 release.

**Comment by localheinz** [comment]
@johnkary 

Also, if you can, please take a look at the ✉️ that I sent you in regard to the new event system of `phpunit/phpunit`!

🤓

**Comment by johnkary** [comment]
@vlaurier Thank you again for your work implementing the Hooks system. I have built on top of your work and introduced PR #83 that completes the Hooks implementation. This will be a new major release version of SpeedTrap once merged.

Please follow progress of #83 for compatibility with PHPUnit 8.0+.

**Pull request closed without merging**